cmd/helm/repo_add.go

```diff
@@ -85,8 +85,8 @@ func (o *repoAddOptions) run(out io.Writer) error {
 		return err
 	}
 
-	// Lock the repository file for concurrent goroutines or processes synchronization
-	fileLock := flock.New(o.repoFile)
+	// Acquire a file lock for process synchronization
+	fileLock := flock.New(o.repoFile[0:len(o.repoFile)-len(filepath.Ext(o.repoFile))] + "lock")
```

Comment:
Non-blocking nit: This might read clearer with a `strings.Replace(o.repoFile, filepath.Ext(o.repoFile), "lock", 1)`

Reply:
I like cleaner code

Reply:
While this may work, it is a little risky doing it this way. `strings.Replace` replaces the first occurrence of the string. If we eventually exposed this as a public API, `repoFile` could be something other than repository.yaml.

in which case one could imagine passing a file called `repository.repo` or something similar, which would end up as `locksitory.repo`.

https://play.golang.org/p/znzd61mfU8l